plugins/history_plugin/history_plugin.cpp

```diff
@@ -493,15 +493,16 @@ namespace eosio {
               ++itr;
             }
 
+
+            const vector<transaction_receipt>* receipts = nullptr;
             auto blk = chain.fetch_block_by_number( result.block_num );
-            if( blk == nullptr ) { // still in pending
-                auto blk_state = chain.pending_block_state();
-                if( blk_state != nullptr ) {
-                    blk = blk_state->block;
-                }
+            if( blk ) {
+               receipts = &blk->transactions;
+            } else if( chain.is_building_block() ) { // still in pending
+               receipts = &chain.get_pending_trx_receipts();
```

Comment:
This seems rather fragile to me. If `get_pending_trx_receipts()` changes this could break easily. How about this:
```
if( blk || chain.is_building_block() ) {
   const vector<transaction_receipt>& receipts = blk ? blk->transactions : chain.get_pending_trx_receipts();
...
```

Reply:
Resolved in https://github.com/EOSIO/eos/pull/6588/commits/5961a38c7abb418eae4342c1f8a6d5e92eb61170.